src/types.ts

```diff
@@ -257,14 +259,25 @@ export const IdentityType = {
         }
     },
 
-    getNewIdentitiesByName: (newIdentitiesByType: IdentitiesByType): IdentitiesByType => {
+    // Strips out functions from Identity Types for easier lookups
+    getValuesAsStrings: (): string[] =>
+        Object.values(IdentityType)
+            .map(value => (isNumber(value) ? value.toString() : undefined))
+            .filter(value => value !== undefined) as string[],
+
+    getNewIdentitiesByName: (
+        newIdentitiesByType: IdentitiesByType
+    ): IdentitiesByType => {
         const newIdentitiesByName: IdentitiesByType = {};
 
         for (const key in newIdentitiesByType) {
-            const identityNameKey = IdentityType.getIdentityName(
-                parseNumber(key)
-            );
-            newIdentitiesByName[identityNameKey] = newIdentitiesByType[key];
+            // IdentityTypes are stored as numbers but are passed in as strings
+            if (IdentityType.getValuesAsStrings().includes(key)) {
```

Comment:
I see what you are doing here.  I don' tknow that this needs to be done _every_ time parseIdentity is called. 
 Is there a way we can prevent that from happening...maybe once upon initiation of the identity module and putting that there as a property?  Otherwise I see a map with an includes and a for loop outside of that.  Seems a _bit_ much?  Although these are limited size arrays/objects and I don't imagine identity calls being made that often.

Just some food for thought.